### PR TITLE
Add max_concurrent_queries_for_all_users setting

### LINKED ISSUE #6636: Add "max_simultaneous_queries_for_all_users" settings.
**Use case**
To set the limit on total number of queries. That limit can be overriden. For example, it can be set to 99 for all users and dba can set it to 100 for itself to run queries even when the server is overloaded.

**Describe alternatives you've considered**
There is global `max_connections` and `max_concurrent_queries` parameters in config. When the server is overloaded, it's impossible to connect and run queries for investigation.

### PR DESCRIPTION
**Pull request by nvartolomei**
Closes #6636.

I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Add max_concurrent_queries_for_all_users setting, see #6636 for use cases.

Detailed description / Documentation draft:

...

By adding documentation, you'll allow users to try your new feature immediately, not when someone else will have time to document it later. Documentation is necessary for all features that affect user experience in any way. You can add brief documentation draft above, or add documentation right into your patch as Markdown files in [docs](https://github.com/ClickHouse/ClickHouse/tree/master/docs) folder.

If you are doing this for the first time, it's recommended to read the lightweight [Contributing to ClickHouse Documentation](https://github.com/ClickHouse/ClickHouse/tree/master/docs/README.md) guide first.

Information about CI checks: https://clickhouse.tech/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Interpreters/ProcessList.cpp [hunk 1]
```diff
@@ -89,6 +89,15 @@ ProcessList::EntryPtr ProcessList::insert(const String & query_, const IAST * as
                 throw Exception("Too many simultaneous queries. Maximum: " + toString(max_size), ErrorCodes::TOO_MANY_SIMULTANEOUS_QUERIES);
         }
 
+        {
+            if (!is_unlimited_query && settings.max_concurrent_queries_for_all_users
+                && processes.size() > settings.max_concurrent_queries_for_all_users)
```
Comment (alexey-milovidov) [inline src/Interpreters/ProcessList.cpp:94]: If I look a few lines above I see that `>=` is used.

File: src/Interpreters/ProcessList.cpp [hunk 2]
```diff
@@ -89,6 +89,15 @@ ProcessList::EntryPtr ProcessList::insert(const String & query_, const IAST * as
                 throw Exception("Too many simultaneous queries. Maximum: " + toString(max_size), ErrorCodes::TOO_MANY_SIMULTANEOUS_QUERIES);
         }
 
+        {
```
Comment (alexey-milovidov) [inline src/Interpreters/ProcessList.cpp:92]: Please add a comment how does it differ from `max_size`.

**Review by alexey-milovidov — COMMENTED** [review]
All other code LGTM.

File: src/Interpreters/ProcessList.cpp (hunk 2 above)
Comment (nvartolomei) [reply src/Interpreters/ProcessList.cpp:92]: Tried my best.

**Pull request merged**